tracer/test/Datadog.Trace.Security.Unit.Tests/IAST/Tainted/VulnerabilityBatchTests.cs

```diff
@@ -0,0 +1,149 @@
+// <copyright file="VulnerabilityBatchTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.Iast;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Security.Unit.Tests.IAST.Tainted;
+
+public class VulnerabilityBatchTests
+{
+    [Fact]
+    public void GivenAnVulnerabilityBatch_WhenAddVulnerability_VulnerabilityIsCorrectlyAdded()
+    {
+        var batch = new VulnerabilityBatch();
+        var source = new Source(2, "name", "value");
+        var source2 = new Source(3, "name2", "value2");
+        var ranges = new Range[] { new Range(0, 2, source), new Range(2, 2, source2) };
+        var evidence = new Evidence("sql_query", ranges);
+        var vulnerability = new Vulnerability("sqli", new Location(), evidence);
+        batch.Add(vulnerability);
+
+        batch.Sources.Count.Should().Be(2);
+        batch.Sources[0].GetInternalId().Should().Be(0);
+        batch.Sources[1].GetInternalId().Should().Be(1);
+
+        batch.Vulnerabilities.Should().BeEquivalentTo(new[]
+        {
+            new
+            {
+                Evidence = new
+                {
+                    ValueParts = new[]
+                    {
+                        new { Source = 0, Value = "sq" },
+                        new { Source = 1, Value = "l_" },
```

Comment:
`source` has an `origin` or `2` and `source2` has an origin of `3`. I don't see that included in the output here. Is that correct? If so, do we actually care about it/need to store it? 🤔

Reply:
Yes, it is correct. The source here (in the valueParts section) is an index that refers to the source list where the origin is actually stored. You can take a look here: https://github.com/DataDog/experimental/blob/main/teams/asm/vulnerability_schema/vulnerability_example.json